bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/helpers/TapoErrorHandler.java

```diff
@@ -90,30 +106,13 @@ public TapoErrorHandler(Exception ex, String infoMessage) {
     /**
      * GET ERROR-MESSAGE
      * 
-     * @param errCode error Number (or constant ERR_CODE )
-     * @return error-message if set constant ERR_CODE_MSG. if not name of ERR_CODE is returned
+     * @param errCode error Number (or constant ERR_API_CODE )
+     * @return error-message if code found in i18n, else return code
      */
     private String getErrorMessage(Integer errCode) {
-        Field[] fields = TapoErrorConstants.class.getDeclaredFields();
-        /* loop ErrorConstants and search for code in value */
-        for (Field f : fields) {
-            String constName = f.getName();
-            try {
-                Integer val = (Integer) f.get(this);
-                if (val != null && val.equals(errCode)) {
-                    Field constantName = TapoErrorConstants.class.getDeclaredField(constName + "_MSG");
-                    String msg = getValueOrDefault(constantName.get(null), "").toString();
-                    if (msg.length() > 2) {
-                        return msg;
-                    } else {
-                        return infoMessage + " (" + constName + ")";
-                    }
-                }
-            } catch (Exception e) {
-                // next loop
-            }
-        }
-        return infoMessage + " (" + errCode.toString() + ")";
+        String key = TapoErrorCode.fromCode(errCode).name().replace("ERR_", "error-").replace("_", "-").toLowerCase();
+        String message = "@text/" + key;
+        return message.replace("%CODE", errCode.toString());
```

Comment:
I do not understand your replace.
Just before message will have for example this value: `@text/error-config-no-bridge`
I can't understand how it could contain `%CODE`

Reply:
I understand what you try to do, it is for `error-unknown`.
You should apply the replace not on the key but on the value returned by i18n request.

Reply:
Instead of `%CODE` you should use `{0]` in the properties file.
And then you have a special syntax to use to provide the parameter. If I correctly remember (to be verified), it should be something like : `@text/error-unknown [ "xxx" ]` where "xxx" is your number.

Reply:
Something like that should work:
```
String message = "@text/" + key;
if (message.contains("{0}") {
    message += " [ \"" + errCode.toString() + "\" ]";
}
```

Reply:
In fact, you could even use {0} in all your i18n entries, rather than having the code hardcoded in the i18n entries.
I mean for example for error-config-no-bridge: `no bridge configured ({0})`
To be tested with at least one error to be sure that the proposed syntax is correct.

Reply:
It's working exactyl as you said. Testet it. Many Thanks again.
And of course in fact i can use `{0}` in any line, but if i there is it necesarry to assign a code to an enum (for debugging, etc)  it would be easyer if it's hardcoded.